keras/models.py

```diff
@@ -32,6 +32,90 @@
     h5py = None
 
 
+def save_mxnet_model(model, prefix, epoch=0):
+    """Save the model as MXNet model. Creates MXNet symbol and params file.
+
+    The saved model can be used to load in MXNet engine for inference. Can be used to perform
+    inference on any language bindings supported by MXNet.
+
+    Note: data_names and data_shapes required in MXNet represent input layer name and shape.
+    When you call save_mxnet_model, these information will be printed.
+    You can change the first dimension of data_shapes to match batch size for inference.
+
+    Below sample code shows how to load a Keras-MXNet pre-trained model saved with 'prefix',
+    input name as '/dense_1_input1' and inference batch size of 1 and data shape of 784.
+
+    ```python
+    >>> import mxnet as mx
+    >>> sym, arg_params, aux_params = mx.model.load_checkpoint(prefix, epoch=0)
+    ##  data_names and data_shapes will be print
+    >>> mod = mx.mod.Module(symbol=sym, data_names=['/dense_1_input1'], \
+                            context=mx.cpu(), label_names=None)
+    >>> mod.bind(for_training=False, data_shapes=[('/dense_1_input1', (1,784))], \
+                 label_shapes=mod._label_shapes)
+    >>> mod.set_params(arg_params, aux_params, allow_missing=True)
+    ##  Prepare data iterator
+    >>> data_iter = mx.io.NDArrayIter(data, label=None, batch_size=1)
+    ##  Predict
+    >>> result = mod.predict(data_iter)
+    ```
+
+    # Arguments
+        model: Keras model instance to be saved as MXNet model.
+        prefix: Prefix name of the saved Model (symbol and params) files.
+                Model will be saved as '<prefix>-symbol.json' and '<prefix>-<epoch>.params'.
+        epoch: (Optional) Tag the params file with epoch of the model being saved. Default is 0.
+               Model params file is saved as '<prefix>-<epoch>.params' or '<prefix>-0000.params' by default.
+
+    # Returns
+        data_names, data_shapes
+    """
+    assert model is not None, 'MXNet Backend: Invalid state. Model cannot be None.'
+
+    # Handle Sequential Model / Functional API Model Case
+    if isinstance(model, Sequential):
+        mxnet_model = model.model
+    elif isinstance(model, Model):
+        mxnet_model = model
+    else:
+        raise ValueError('MXNet Backend: Invalid model type. Supported Models - Sequential, Model.')
+
+    assert mxnet_model is not None, 'MXNet Backend: Invalid state. MXNet Model cannot be None.'
+
+    if not mxnet_model.compiled:
+        raise ValueError('MXNet Backend: Model is not compiled. Cannot save the MXNet model!')
+
+    # Saving MXNet model for Inference in native MXNet engine.
+    symbol = mxnet_model._pred_mxnet_symbol
```

Comment:
I had a question here - are we sure we would only be saving an MXNet model for inference? Does it make sense to have an option to control which bucket to return?

Reply:
Thanks for reviewing Dhruv. 
1. In Keras-MXNet, we maintain 3 buckets - train, test, pred. Each having symbol of their own, but they all share the parameters.
2. So after training pred bucket is bound to symbol that can be used for inference.
3. Allowing 'train' and 'test' symbols to be saved may not be very useful at this point of time, because they contain few keras specific parameters and symbols. However, that is a good suggestion, can you let me know any usecase for this feature you are thinking?

Reply:
@sandeep-krishnamurthy That's fair. Im thinking of the usecase, researcher trains on keras-mxnet, hands-off a layer to production pipeline in symbol, they freeze that layer and continue to train the larger network. But its a very niche thing - so I just wanted to see if it was easy, what you did makes a lot of sense!

Reply:
Thats a very valid usecase. Let me create a Github issue to track this feature request. https://github.com/awslabs/keras-apache-mxnet/issues/81